src/Inventory/Asset/Device.php

```diff
@@ -127,7 +127,7 @@ public function handle()
                             break;
                         }
                         $compare = explode(':', $compare);
-                        if (!isset($i_input[$field]) && !isset($existing_item[$field])) {
+                        if (!isset($i_input[$field]) || !isset($existing_item[$field])) {
```

Comment:
I am not sure of that, but I think that when a comparison field is present in one item but not in the other, it probably means that items are different, and therefore `$equals` should be set to false.
```php
if (!isset($i_input[$field]) && !isset($existing_item[$field])) {
    //field not present, skip
    continue;
} elseif (
    (isset($i_input[$field]) && !isset($existing_item[$field]))
    || (!isset($i_input[$field]) && isset($existing_item[$field]))
) {
    // field is present in an item but absent in the other, it cannot be equal
    $equals = false;
    continue;
}
```

Reply:
>```php
> } elseif (
>     (isset($i_input[$field]) && !isset($existing_item[$field]))
>     || (!isset($i_input[$field]) && isset($existing_item[$field]))
> ) {
>     // field is present in an item but absent in the other, it cannot be equal
>```


Too many checks. :) Second case is only possible when all or some items are not set.

```php
if (!isset($i_input[$field]) && !isset($existing_item[$field])) {
    //field not present, skip
    continue;
} elseif (!isset($i_input[$field]) || !isset($existing_item[$field])) {
    // field is present in an item but absent in the other, it cannot be equal
    $equals = false;
    continue;
}
```